File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    memo = {}

    # array에 있는 원소들을 순회하면서 카운팅을 함
    # num이 없으면 애초에 카운팅이 x
    for num in array:
        memo[num] = memo.get(num, 0) + 1

    # 최빈값의 빈도 계산
    # items를 순회하는데, value가 내가 찾는 max_count와 일치하면 key를 result 리스트에 append
    max_count = max(memo.values())

    result_count = 0
    mode = -1

    for k, v in memo.items():
        if v == max_count:
            result_count += 1
            mode = k
    # corrected using ternary operator
    return -1 if result_count > 1 else mode

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <unordered_map> // 딕셔너리 대용
// #include <iostream>
using namespace std;

int solution(vector<int> array)
{
    unordered_map<int, int> memo;

    // 배열의 요소를 순회하면서 <K, V>를 저장
    for(auto& num: array) {
        memo[num]++; // dict.get(key, 0) + 1와 비슷하게 동작
    }   
    
    // 최댓값 계산
    int max_count = 0;
    for (const auto& pair: memo) {
        max_count = max(max_count, pair.second); // value의 최댓값을 저장
    }

    // 최빈값 추출
    int result_count = 0, mode = -1;

    for (const auto& pair: memo) {
        if (pair.second == max_count) {
            result_count++;
            mode = pair.first;
        }
    }
    // 최빈값이 여러 개인지 판단
    return (result_count > 1) ? -1 : mode;
}